commands/upgrade.ts

```diff
@@ -0,0 +1,107 @@
+import { onCancel, promptSource } from '../utils/share'
+
+const transformerDirectory = join(__dirname, '../', 'transforms')
+
+export async function upgrade(source?: string) {
+  const sourceSelected = source || (await promptSource('Which directory should the codemods be applied to?'))
+
+  if (!sourceSelected) {
+    console.info('> Source path for project is not selected. Exits the program. \n')
+    process.exit(1)
+  }
+
+  try {
+    const packageJsonPath = resolve(sourceSelected || '', 'package.json')
+    const packageJson = JSON.parse(await readFile(packageJsonPath, 'utf8'))
+
+    const codemods = suggestCodemods(packageJson)
+
+    if (codemods.length === 0) {
+      console.info('> No codemods are suggested for this project. \n')
+
+      return
+    }
+
+    const { codemodsSelected } = await prompts(
+      {
+        type: 'multiselect',
+        name: 'codemodsSelected',
+        message: `The following 'codemods' are recommended for your upgrade. Select the ones to apply.`,
+        choices: codemods.map(({ description, value, version }) => {
+          return {
+            title: `(v${version}) ${value}`,
+            description,
+            value,
+            selected: true,
+          }
+        }),
+      },
+      { onCancel },
+    )
+
+    const args: Options = {
+      dry: false,
+      babel: false,
+      silent: true,
+      ignorePattern: '**/node_modules/**',
+      extensions: 'cts,mts,ts,js,mjs,cjs',
+    }
+
+    const results = {
+      ok: 0,
+      skipped: 0,
+      failed: 0,
+      unmodified: 0,
+    }
+
+    for (const codemod of codemodsSelected) {
+      const transformerPath = require.resolve(`${transformerDirectory}/${codemod}.js`)
+
+      console.log(`> Applying codemod: ${codemod}`)
+      const { ok, skip, error, nochange } = await jscodeshift(transformerPath, [resolve(sourceSelected)], args)
+
+      results.ok += ok
+      results.skipped += skip
+      results.failed += error
+      results.unmodified += nochange
+    }
+
+    console.log('\n> Summary of the upgrade')
+    console.log(`> ${results.ok} codemods were applied successfully`)
+    console.log(`> ${results.skipped} codemods were skipped`)
+    console.log(`> ${results.failed} codemods failed`)
+    console.log(`> ${results.unmodified} codemods were skipped because they didn't change anything`)
+  } catch (err) {
+    if (err.code === 'ENOENT') {
+      console.info('> No package.json found in the selected directory. \n')
+      process.exit(1)
+    } else {
+      console.error(err.message)
+    }
+  }
+}
+
+function suggestCodemods(packageJson) {
+  const { dependencies } = packageJson
+
+  if (dependencies?.express == null) {
+    console.info('> No express dependency found in package.json. \n')
+
+    process.exit(0)
+  }
+
+  const expressVersion = coerce(dependencies.express)?.version ?? '4.0.0'
+
+  const codemodsSuggested = TRANSFORM_OPTIONS.filter((a) => {
+    return compare(a.version, expressVersion) > 0
```

Comment:
shouldn't we apply our codemods only if the project is on v4?

according to docs:

> compare(v1, v2): Return 0 if v1 == v2, or 1 if v1 is greater, or -1 if v2 is greater.

Reply:
It is more thought out so that in the future more codemods can be added as Express evolves, and not limit us to version 4